src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -3897,15 +3897,15 @@ private static String ExpandPredefinedFormat(ReadOnlySpan<char> format, ref Date
                         {
                             if (!MatchAbbreviatedMonthName(ref str, dtfi, ref tempMonth))
                             {
-                                result.SetFailure(ParseFailureKind.Format, nameof(SR.Format_BadDateTime), null);
+                                result.SetFailure(ParseFailureKind.FormatWithOriginalDateTime, nameof(SR.Format_BadDateTime));
                                 return (false);
                             }
                         }
                         else
                         {
                             if (!MatchMonthName(ref str, dtfi, ref tempMonth))
                             {
-                                result.SetFailure(ParseFailureKind.Format, nameof(SR.Format_BadDateTime), null);
+                                result.SetFailure(ParseFailureKind.FormatWithOriginalDateTime, nameof(SR.Format_BadDateTime));
```

Comment:
There are 103 occurrences of this line `result.SetFailure(ParseFailureKind.FormatWithOriginalDateTime, nameof(SR.Format_BadDateTime));`

What about adding a method to DateTimeResult so it becomes just `result.SetBadDateTimeFailure()` or suchlike? That would likely improve code size.

Reply:
Similarly you could replace this pattern with one call
```c#
                   result.SetFailure(ParseFailureKind.FormatWithFormatSpecifier, nameof(SR.Format_BadFormatSpecifier));
                   result.failedFormatSpecifier = formatParam;
```
There's only a few of those but it would make it impossible to forget to set formatParam when you use FormatWithFormatSpecifier